gax/src/main/java/com/google/api/gax/tracing/TracedServerStreamingCallable.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * <p>This class is meant to be an internal implementation google-cloud-java clients only.
+ */
+@BetaApi("The surface for tracing is not stable and might change in the future")
+@InternalApi("For internal use by google-cloud-java clients only")
+public class TracedServerStreamingCallable<RequestT, ResponseT>
+    extends ServerStreamingCallable<RequestT, ResponseT> {
+
+  @Nonnull private final ApiTracerFactory tracerFactory;
+  @Nonnull private final SpanName spanName;
+  @Nonnull private final ServerStreamingCallable<RequestT, ResponseT> innerCallable;
+
+  public TracedServerStreamingCallable(
+      @Nonnull ServerStreamingCallable<RequestT, ResponseT> innerCallable,
+      @Nonnull ApiTracerFactory tracerFactory,
+      @Nonnull SpanName spanName) {
+    this.tracerFactory = Preconditions.checkNotNull(tracerFactory, "tracerFactory can't be null");
+    this.spanName = Preconditions.checkNotNull(spanName, "spanName can't be null");
+    this.innerCallable = Preconditions.checkNotNull(innerCallable, "innerCallable can't be null");
+  }
+
+  @Override
+  public void call(
+      RequestT request, ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
+
+    ApiTracer tracer = tracerFactory.newTracer(spanName);
+    TracedResponseObserver<ResponseT> tracedObserver =
+        new TracedResponseObserver<>(tracer, responseObserver);
+
+    context = context.withTracer(tracer);
+
+    try {
+      innerCallable.call(request, tracedObserver, context);
+    } catch (RuntimeException e) {
+      tracedObserver.onError(e);
+      throw e;
+    }
+  }
+
+  /** A {@link ResponseObserver} to mark a started operation trace as finished. */
+  private static class TracedResponseObserver<ResponseT> implements ResponseObserver<ResponseT> {
+    private final ApiTracer tracer;
+    private final ResponseObserver<ResponseT> innerObserver;
+    private volatile boolean wasCancelled;
+
+    private TracedResponseObserver(
+        @Nonnull ApiTracer tracer, @Nonnull ResponseObserver<ResponseT> innerObserver) {
+      this.tracer = Preconditions.checkNotNull(tracer, "tracer can't be null");
+      this.innerObserver = Preconditions.checkNotNull(innerObserver, "innerObserver can't be null");
+    }
+
+    @Override
+    public void onStart(final StreamController controller) {
+      innerObserver.onStart(
+          new StreamController() {
+            @Override
+            public void cancel() {
+              wasCancelled = true;
+              controller.cancel();
+            }
+
+            @Override
+            public void disableAutoInboundFlowControl() {
+              controller.disableAutoInboundFlowControl();
+            }
+
+            @Override
+            public void request(int count) {
+              controller.request(count);
+            }
+          });
+    }
+
+    @Override
+    public void onResponse(ResponseT response) {
+      tracer.responseReceived();
```

Comment:
Same ordering-specific question as in another PR (BTW, Ideally we would want this code to be shared, please see the above coment)

Reply:
From [bidi PR](https://github.com/googleapis/gax-java/pull/636/files#r251551809)
Does the order of calls make any difference here (or can make any at least potentially in a corner case)? It seems logical to report response received (i.e. call tracer) as a last statement, so the reporting is done when everything is really done.

Reply:
The span should only wrap the work that the client/callable is doing, not the user's callbacks. Ie. if someone does this:

```java
client.readRowsCallable().call(request, new ResponseObserver() {
  void onResponse(Row row) {
     Thread.sleep(1000)
  }
}
```
We don't want to include the 1 second sleep. That should be part of the caller's span not the clients